src/index.js

```diff
@@ -1,21 +1,33 @@
 "use strict";
 
-let WritableStream = require("./writable_stream");
+const WritableStream = require("./writable_stream");
 
-module.exports = class Renderer {
-	constructor(bundlePath) {
-		this.bundlePath = bundlePath;
-		this.reload();
-	}
+module.exports = rendererPath => {
+	return (req, res, next) => {
+		const renderer = typeof rendererPath === "function" ?
+			rendererPath :
+			resolveRenderer(res.app, rendererPath, { cache: true });
 
-	render(response, tag, params) {
-		// TODO: `response.status(200);`?
-		let stream = new WritableStream(response);
-		this._render(stream, tag, params);
-		response.end();
-	}
+		res.complate = (tag, params) => {
+			let stream = new WritableStream(res);
+			renderer(stream, tag, params);
+			res.end();
+		};
 
-	reload() {
-		this._render = require(this.bundlePath);
-	}
+		next();
+	};
 };
+
+function resolveRenderer(app, rendererPath, { cache } = {}) {
+	// by default the require cache is disabled for dev environment,
+	// so the given renderer is updated on each request.
+	// This enables a watch/compile setup for re-creating the renderer
+	// and its components.
+	//
+	// Set `cache` to true, if you want to cache the renderer also in dev-mode.
+	if(!cache && !app.enabled("view cache")) {
+		delete require.cache[rendererPath];
+	}
+
+	return require(rendererPath);
+}
```

Comment:
hmm, while that is more readable yes, it looses some options I think a worth the hassle.

1. it does not allow to inject the bundle directly anymore, which decreases the possibilities of the user and decreases testing capabilities IMHO

2. cache configuration is missing. While I see the point, that the `require.cache` override is useful, it is quire intrusive, and it may be a good thing (at least for debugging purpose) to be able to skip it.

3. it requires the file twice if the cache is disabled